src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -63,11 +64,13 @@ export default abstract class QueryBase implements QueryInterface {
    * Retrieves the current gas costs of performing a fillRelay contract at the referenced Spoke Pool
    * @returns The gas estimate for this function call (multplied with the optional buffer)
    */
-  async getGasCosts(): Promise<BigNumberish> {
-    const tx = await createUnsignedFillRelayTransaction(this.spokePool, this.usdcAddress, this.simulatedRelayerAddress);
+  async getGasCosts(deposit: Deposit, relayAddress?: string): Promise<BigNumberish> {
+    const relayerToSimulate = relayAddress ?? this.simulatedRelayerAddress;
+    const correspondingFill = buildFillForSimulatingFullDeposit(deposit, relayerToSimulate);
```

Comment:
Note: I am verbose here so that this function can be used to simulate other fills in the future.